pkg/apiserver/domain/service/workflow.go

```diff
@@ -358,45 +358,63 @@ func (w *workflowServiceImpl) SyncWorkflowRecord(ctx context.Context) error {
 			Name:      appName,
 			Namespace: record.Namespace,
 		}, app); err != nil {
+			if apierrors.IsNotFound(err) {
+				if err := w.setRecordToTerminated(ctx, record.AppPrimaryKey, record.Name); err != nil {
+					log.Logger.Errorf("failed to set the record status to terminated %s", err.Error())
+				}
+				continue
+			}
 			klog.ErrorS(err, "failed to get app", "oam app name", appName, "workflow name", record.WorkflowName, "record name", record.Name)
 			continue
 		}
 
+		// there is a ":" in the default app revision
+		recordName := strings.Replace(app.Status.Workflow.AppRevision, ":", "-", 1)
+
 		// try to sync the status from the running application
-		if app.Annotations != nil && app.Status.Workflow != nil && app.Status.Workflow.AppRevision == record.Name {
-			if err := w.syncWorkflowStatus(ctx, app, record.Name, app.Name); err != nil {
+		if app.Annotations != nil && app.Status.Workflow != nil && recordName == record.Name {
+			if err := w.syncWorkflowStatus(ctx, record.AppPrimaryKey, app, record.Name, app.Name); err != nil {
 				klog.ErrorS(err, "failed to sync workflow status", "oam app name", appName, "workflow name", record.WorkflowName, "record name", record.Name)
 			}
 			continue
 		}
 
 		// try to sync the status from the controller revision
-		cr := &appsv1.ControllerRevision{}
-		if err := w.KubeClient.Get(ctx, types.NamespacedName{
-			Name:      fmt.Sprintf("record-%s-%s", appName, record.Name),
-			Namespace: record.Namespace,
-		}, cr); err != nil {
-			klog.ErrorS(err, "failed to get controller revision", "oam app name", appName, "workflow name", record.WorkflowName, "record name", record.Name)
+		var revision = &model.ApplicationRevision{AppPrimaryKey: record.AppPrimaryKey, Version: record.RevisionPrimaryKey}
+		if err := w.Store.Get(ctx, revision); err != nil {
+			log.Logger.Errorf("failed to get the application revision from database %s", err.Error())
```

Comment:
Same `setRecordTerminate` here?

Reply:
If the application revision is not found, set the record status to failure.